interop-tests/Dockerfile

```diff
@@ -0,0 +1,20 @@
+# syntax=docker/dockerfile:1.5-labs
+FROM rust:1.67.0
+
+# Setup protoc. TODO this breaks reproducibility and uses an old version of protoc.
+# In the future protobuf generated files will be checked in, so we can remove this
+WORKDIR /protoc-setup
+RUN apt-get update && apt-get install -y cmake protobuf-compiler
+
+# Run with access to the target cache to speed up builds
+WORKDIR /workspace
+ADD . .
+RUN --mount=type=cache,target=./target \
+    --mount=type=cache,target=/usr/local/cargo/registry \
```

Comment:
What exactly is the cache referenced here? Old layers that don't match exactly?

Reply:
This is a fairly new cool feature. It is a cache that is saved separate from layers. So even if the earlier layers get rebuilt we still persist our cargo and build cache. So builds will be just be fast without any tricks. https://docs.docker.com/engine/reference/builder/#run---mount

Reply:
This is amazing! Thank for your introducing me to this feature, that is indeed a godsend for Rust-based containers!